packages/dev/core/src/Layers/layerSceneComponent.ts

```diff
@@ -95,40 +101,58 @@ export class LayerSceneComponent implements ISceneComponent {
         }
     }
 
-    private _drawCameraPredicate(layer: Layer, isBackground: boolean, cameraLayerMask: number): boolean {
-        return !layer.renderOnlyInRenderTargetTextures && layer.isBackground === isBackground && (layer.layerMask & cameraLayerMask) !== 0;
+    private _drawCameraPredicate(layer: Layer, isBackground: boolean, applyPostProcess: boolean, cameraLayerMask: number): boolean {
+        return (
+            !layer.renderOnlyInRenderTargetTextures &&
+            layer.isBackground === isBackground &&
+            layer.applyPostProcess === applyPostProcess &&
+            (layer.layerMask & cameraLayerMask) !== 0
+        );
     }
 
     private _drawCameraBackground(camera: Camera): void {
         this._draw((layer: Layer) => {
-            return this._drawCameraPredicate(layer, true, camera.layerMask);
+            return this._drawCameraPredicate(layer, true, true, camera.layerMask);
         });
     }
 
-    private _drawCameraForeground(camera: Camera): void {
+    private _drawCameraForegroundWithPostProcessing(camera: Camera): void {
         this._draw((layer: Layer) => {
-            return this._drawCameraPredicate(layer, false, camera.layerMask);
+            return this._drawCameraPredicate(layer, false, true, camera.layerMask);
         });
     }
 
-    private _drawRenderTargetPredicate(layer: Layer, isBackground: boolean, cameraLayerMask: number, renderTargetTexture: RenderTargetTexture): boolean {
+    private _drawCameraForegroundWithoutPostProcessing(camera: Camera): void {
+        this._draw((layer: Layer) => {
+            return this._drawCameraPredicate(layer, false, false, camera.layerMask);

```

Comment:
is it really false, false ? I think I'm confused by the goal of this PR :)

Reply:
For example, here, if layer.isBackground === false and layer.applyPP = false then the PP WILL be applied right?